fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+this.tenantsTable = Table.builder(adminSchemaName, TENANTS)
+    .setVersion(1)
+    .addIntColumn(            MT_ID,             false)
+    .addVarcharColumn(      TENANT_NAME,        36,  false)
+    .addVarcharColumn(    TENANT_STATUS,        16,  false)
+    .addUniqueIndex(IDX + "TENANT_TN", TENANT_NAME)
+    .addPrimaryKey("TENANT_PK", MT_ID)
+    .setTablespace(fhirTablespace)
+    .build(model);
+```
+
+If this table is updated, the changes must be manually apply the migration steps to the `TENANTS` and the corresponding indices. 
+
+**Table: TENANT_KEYS**
+
+The Tenant Keys table stores a hashed version of the tenant specific key. Upon creating a session and calling the SET_TENANT procedure, the SET_TENANT queries the incoming key against the TENANT_KEYS database to map to a specific tenant MT_ID. 
+
+`AdminSchemaGenerator.addTenantKeysTable` method adds the TENANT_KEYS table, and updates to this table definition requries a change to the setVersion (incrementing by 1).
+
+``` java
+this.tenantKeysTable = Table.builder(adminSchemaName, TENANT_KEYS)
+    .addIntColumn(        TENANT_KEY_ID,             false)
+    .addIntColumn(                MT_ID,             false)
+    .addVarcharColumn(      TENANT_SALT,        44,  false)
+    .addVarbinaryColumn(    TENANT_HASH,        32,  false)
+    .addUniqueIndex(IDX + "TENANT_KEY_SALT", TENANT_SALT)
+    .addUniqueIndex(IDX + "TENANT_KEY_TIDH", MT_ID, TENANT_HASH)
+    .addPrimaryKey("TENANT_KEY_PK", TENANT_KEY_ID)
+    .addForeignKeyConstraint(FK + TENANT_KEYS + "_TNID", adminSchemaName, TENANTS, MT_ID)
+    .setTablespace(fhirTablespace)
+    .build(model);
+```
+
+If this table is updated, the changes must be manually apply the migration steps to the `TENANT_KEYS` and the corresponding indices. 
+
+If the UniqueIndex or ForeignKeyConstraint is removed, the object must be removed. 
+
+## Managing PROCEDURE
+
+`FHIR_ADMIN.set_tenant` is created in the FHIR_ADMIN schema.  The stored procedure is a 
+
+If updates are made to the `set_tenant.sql`, the `FhirSchemaGenerator.buildAdminSchema`'s FhirSchemaConstants.INITIAL_VERSION must be updated to the next highest integer (in this case 2). 
+
+``` java
+// The set_tenant procedure can be created after all the admin tables are done
+ProcedureDef setTenant = model.addProcedure(this.adminSchemaName, 
+    SET_TENANT, 
+    FhirSchemaConstants.INITIAL_VERSION, 
+    () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, adminSchemaName, SET_TENANT.toLowerCase() + ".sql", null), 
+    Arrays.asList(allAdminTablesComplete), 
+    procedurePrivileges);
+        setTenant.addTag(SCHEMA_GROUP_TAG, ADMIN_GROUP);
+```
+
+When the `fhir-persistence-schema` updateProcedure is executed, the READ grant is applied again, and the procedure is updated. The Procedure privilege is reset upon re-executing the action. 
+
+If you change the stored procedure signature, you MUST drop the stored procedure, before applying the stored procedure to the database. 
```

Comment:
```suggestion
If you change the stored procedure signature, you MUST drop the stored procedure before applying the stored procedure to the database. 
```

Reply:
Also, is this different than any other stored procedure?
Why is it that it must be dropped instead of updated/replaced in-place?

Reply:
you may recall we hit this before.  If the method has say three input, and you try dropping it, you must use the specific signature to drop it.  The code currently tries to create a new unique method, leaving two in place, not one.